app/services/defer_participant.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+class DeferParticipant
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :cpd_lead_provider
+  attribute :participant_id
+  attribute :reason
+  attribute :course_identifier
+
+  validates :cpd_lead_provider,
+            induction_record: true
+  validates :reason,
+            presence: { message: I18n.t(:missing_reason) },
+            inclusion: {
+              in: ParticipantProfile::DEFERRAL_REASONS,
+              message: I18n.t(:invalid_reason),
+            }
+  validates :course_identifier, course: true
+  validate :not_already_deferred
+  validate :not_already_withdrawn
+
+  def call
+    ActiveRecord::Base.transaction do
+      create_deferred_participant_profile_state!
+      create_deferred_induction_record!
+      participant_profile.training_status_deferred!
+    end
+
+    participant_profile.record_to_serialize_for(lead_provider: cpd_lead_provider.lead_provider)
+  end
+
+  def participant_identity
+    @participant_identity ||= ParticipantIdentity.find_by(external_identifier: participant_id)
+  end
+
+  def participant_profile
+    @participant_profile ||=
+      ParticipantProfileResolver.call(
+        participant_identity:,
+        course_identifier:,
+        cpd_lead_provider:,
+      )
+  end
+
+  def schedule
```

Comment:
Is this being used anywhere?

Reply:
We need the surface schedule for the induction record validator, see: https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/validators/induction_record_validator.rb#L31